shop/rest/renderers.py

```diff
@@ -3,6 +3,46 @@
 
 from rest_framework import renderers
 from rest_framework.compat import template_render
+from rest_framework.exceptions import APIException
+
+from shop.models.cart import CartModel
+from shop.serializers.cart import CartSerializer
+
+
+class ShopTemplateHTMLRenderer(renderers.TemplateHTMLRenderer):
+    """
+    Modified TemplateHTMLRenderer, which shall be used to render templates used by django-SHOP.
+    Instead of polluting the template context with the serialized data, that information is
+    stored inside a separate `data` attribute, which allows to add a Cart and Paginator object.
+
```

Comment:
Each DRF Renderer class has a method named ``get_template_names``, hence it is perfectly possible, maybe even intended to created hybrid templates, such as required for Django-CMS page view or other Django Views.

If we merge the serialized context on the root level of the scope, we pollute the root scope and definitely risk to overwrite important attributes.

Currently templates for the ``ShopTemplateHTMLRenderer `` are compatible with the ``CMSPageRenderer `` which has proven to be mature and is being used for almost two years now.

Context processors are applied globally, so I don't see any benefit (unless I see a proposal on how they would improve compatibility).

Again, all HTML templates for django-SHOP are intended to be rendered by one of those two renderers ``CMSPageRenderer `` and ``ShopTemplateHTMLRenderer ``. If such a use pattern is inappropriate, please point me onto the documentation stating so.